code/Test_definitions/DeviceIdentifier_retrieve_identifier.feature

```diff
@@ -0,0 +1,141 @@
+    When the HTTPS "POST" request is sent
+    And the request body has the field ipv4Address.publicAddress with a value of PUBLICIPV4ADDRESS
+    And the request body has the field ipv4Address.publicPort with a value of PUBLICPORT
+    And the response header "x-correlator" has same value as the request header "x-correlator"
+    And the response header "Content-Type" is "application/json"
+    And the response body complies with the OAS schema at "/components/schemas/200RetrieveIdentifier"
+    Then the response status code is 200
+    And the response property "$.imeisv" is IMEISV2
+
+  @DeviceIdentifier_retrieve_identifier200_missing_device_information
+  Scenario:  retrieve identifier but no device information in request
+    Given they use the base url
+    And the resource is "/retrieve-identifier"
+    And one of the scopes associated with the access token is device-identifier:retrieve-identifier
+    When the HTTPS "POST" request is sent
+    And the connection the request is sent over originates from a device with PHONENUMBER1
+    And the request body has NO the field phoneNumber nor other device information
+    And the response header "x-correlator" has same value as the request header "x-correlator"
+    And the response header "Content-Type" is "application/json"
+    And the response body complies with the OAS schema at "/components/schemas/ErrorResponse"
+    Then the response status code is 400
+    And the response property "$.status" is 400
+    And the response property "$.code" is "INVALID_ARGUMENT"
+    And the response property "$.message" contains a user friendly text
+
+  @DeviceIdentifier_retrieve_identifier201_missing_scope
+  Scenario:  retrieve device identifier with valid access token but scope device-identifier:retrieve-identifier is missing
+    Given they use the base url
+    And the resource is "/retrieve-identifier"
+    And none of the scopes associated with the access token is device-identifier:retrieve-identifier
+    When the HTTPS "POST" request is sent
+    And the connection the request is sent over originates from a device with PHONENUMBER1
+    And the request body has the field phoneNumber with a value of PHONENUMBER1
+    And the response header "x-correlator" has same value as the request header "x-correlator"
+    And the response header "Content-Type" is "application/json"
+    And the response body complies with the OAS schema at "/components/schemas/ErrorResponse"
+    Then the response status code is 401
+    And the response property "$.status" is 401
+    And the response property "$.code" is "UNAUTHENTICATED"
+    And the response property "$.message" is "Request not authenticated due to missing, invalid, or expired credentials."
+
+  @DeviceIdentifier_retrieve_identifier202_expired_access_token
+  Scenario:  retrieve device identifier with expired access token
+    Given they use the base url
```

Comment:
```suggestion
    Given the correct base url for the API provider is used
```

Reply:
I used the same text as other non-DI test definitions...
Use apiRoot like  in https://github.com/camaraproject/DeviceIdentifier/blob/main/code/API_definitions/CAMARA%20Mobile%20Device%20Identifier%20API.yaml#L136 ?